aten/src/ATen/cuda/CUDAGraph.cpp

```diff
@@ -55,6 +59,29 @@ CaptureId_t capture_sequence_id() {
  * describes memory management for captures.
  */
 
+int CUDAGraph::pending_event_queries = 0;
```

Comment:
It would be easier to make it atomic, so you do not need to store mutex object and lock/unlock it every time you modify it.